pkg/operator/staticresource/controller.go

```diff
@@ -0,0 +1,289 @@
+	"sigs.k8s.io/controller-runtime/pkg/predicate"
+	"sigs.k8s.io/controller-runtime/pkg/reconcile"
+	"sigs.k8s.io/controller-runtime/pkg/source"
+)
+
+const (
+	controllerName    = "staticresource"
+	deploymentName    = "cloud-credential-operator"
+	operatorNamespace = "openshift-cloud-credential-operator"
+)
+
+var (
+	staticFiles = []string{
+		"v4.1.0/aws-pod-identity-webhook/sa.yaml",
+		"v4.1.0/aws-pod-identity-webhook/clusterrole.yaml",
+		"v4.1.0/aws-pod-identity-webhook/role.yaml",
+		"v4.1.0/aws-pod-identity-webhook/clusterrolebinding.yaml",
+		"v4.1.0/aws-pod-identity-webhook/rolebinding.yaml",
+		"v4.1.0/aws-pod-identity-webhook/svc.yaml",
+	}
+	templateFiles = []string{
+		"v4.1.0/aws-pod-identity-webhook/deployment.yaml",
+		"v4.1.0/aws-pod-identity-webhook/mutatingwebhook.yaml",
+	}
+	defaultCodecs = serializer.NewCodecFactory(scheme.Scheme)
+	defaultCodec  = defaultCodecs.UniversalDeserializer()
+)
+
+type staticResourceInitialApplyController struct {
+	reconciler *staticResourceReconciler
+}
+
+func (c *staticResourceInitialApplyController) Start(stopCh <-chan struct{}) error {
+	c.reconciler.ReconcileResources()
+	<-stopCh
+	return nil
+}
+
+func Add(mgr manager.Manager, kubeconfig string) error {
+	// Currently the only static assets are AWS only
+	infraStatus, err := platform.GetInfraStatusUsingKubeconfig(mgr, kubeconfig)
+	if err != nil {
+		return err
+	}
+	platformType := platform.GetType(infraStatus)
+	if platformType != configv1.AWSPlatformType {
+		return nil
+	}
+
+	log.Infof("Setting up static resource controller. Platform Type is %s", platformType)
+
+	config := mgr.GetConfig()
+	clientset, err := kubernetes.NewForConfig(config)
+	if err != nil {
+		return err
+	}
+
+	controllerRef := &corev1.ObjectReference{
+		Kind:      "deployment",
+		Namespace: operatorNamespace,
+		Name:      deploymentName,
+	}
+	eventRecorder := events.NewKubeRecorder(clientset.CoreV1().Events(operatorNamespace), deploymentName, controllerRef)
+
+	r := &staticResourceReconciler{
+		clientset:     clientset,
+		logger:        log.WithFields(log.Fields{"controller": controllerName}),
+		eventRecorder: eventRecorder,
+	}
+
+	c, err := controller.New(controllerName, mgr, controller.Options{Reconciler: r})
+	if err != nil {
+		return err
+	}
+
+	p := predicate.Funcs{
+		UpdateFunc: func(e event.UpdateEvent) bool {
+			return isManaged(e.MetaNew)
+		},
+		CreateFunc: func(e event.CreateEvent) bool {
+			return isManaged(e.Meta)
+		},
+		DeleteFunc: func(e event.DeleteEvent) bool {
+			return isManaged(e.Meta)
+		},
+	}
+
+	cache := mgr.GetCache()
+	allFiles := append(staticFiles, templateFiles...)
+	for _, file := range allFiles {
+		objBytes := v410_00_assets.MustAsset(file)
+		obj, _, err := defaultCodec.Decode(objBytes, nil, nil)
+		if err != nil {
+			return err
+		}
+
+		informer, err := cache.GetInformer(obj)
+		if err != nil {
+			return err
+		}
+
+		err = c.Watch(&source.Informer{Informer: informer}, &handler.EnqueueRequestForObject{}, p)
```

Comment:
This is a neat approach to watching everything we created, we may use this in Hive where our operator currently does not watch everything.

Any concerns about memory now that we're watching and thus caching a number of new types cluster wide?

Reply:
I do have concerns about memory and about privilege.  We have to put these abilities on a `ClusterRole` rather than a `Role`, which is not ideal.

I've been looking for a way to limit the scope of the watch to be namespace local, but haven't found it yet.

Reply:
@staebler sorry to pull you in here but where did we land with per-namespace watches and limiting cache, if memory serves there was no way to do this?

Reply:
So the cache is at the Manager level, not on the Controller. As other parts of the operator do watch things on a cluster wide basis (namespace creation, secrets, etc), we can't scope down the manager cache.

We could create a new controller-local cache, separate from the Manager cache, and base the Watches on it.  I'll play around with it.